charts/tekton-pipelines/values.yaml

```diff
@@ -366,6 +438,43 @@ buildpacks:
 kaniko:
   # -- should we enable the kaniko pipeline
   enabled: false
+  postDeployTaskSteps:
+    - name: argo-events
+      image: badouralix/curl-jq
+      imagePullPolicy: IfNotPresent
+      resources: {}
+      script: |
+        #!/usr/bin/env sh
+
+        # add extra params from env and convert them to lowercase to work with this
+        # data later in argo workflow
+        EXTRA=$(jq -n env | jq 'walk(if type=="object" then with_entries(.key|=ascii_downcase) else . end)')
```

Comment:
IMHO, we could simplify it if we just use another line in trigger binding and use exactly as other parameters:
```bash
JSON_TEMPLATE='{
  "project": "$(params.project)",
  "environment": "$(params.environment)",
  "application": "$(params.application)",
  "sha": "$(params.sha)",
  "extra": "$(params.post_build_step_extra)"
}'
```
Just no need for jq, variable parsing, transformation, printf, etc. => simpler to understand.

Reply:
Not really, that was done for a reason. And there are few restrictions:

1. I can't add json / map variables to Tekton vars
2. I wanted to make this approach common for different behaviors (to not hard code vars for `github-dispatch` Sensor only)

So basically I can have lots of different argo-events sensors for the same event `build-finished`, each of them may need different params. For example in my case I need a `dispatch_repository` for `github-dispatch` Sensor (which is run by post-deploy task), but you may have any other Sensor connected to `post-deploy` event with your custom params to trigger any other behavior. So with this implementation we don't need to update TriggerBinding or build-config ConfigMaps in `saritasa-devops-helm-charts` all the time when we add new Sensors to satisfy their params.

So basically, I have a common `post-deploy` task, which just sends `build-finished` webhook to argo-events and passes all default params from TriggerBinding and extra params from  `*-build-pipeline-config` ConfigMap and this gives me opportunity to avoid changing our tekton helm charts all the time for new sensors.